pkg/api/persistentvolume/util.go

```diff
@@ -127,6 +127,18 @@ func VisitPVSecretNames(pv *api.PersistentVolume, visitor Visitor) bool {
 				return false
 			}
 		}
+	case source.Glusterfs != nil:
+		if source.Glusterfs.EndpointsNamespace != nil {
+			// previously persisted PV objects use claimRef namespace
+			ns := getClaimRefNamespace(pv)
+			if len(*source.Glusterfs.EndpointsNamespace) > 0 {
+				// use the endpoint namespace if namespace is set
+				ns = *source.Glusterfs.EndpointsNamespace
+			}
+			if !visitor(ns, *source.Glusterfs.EndpointsNamespace, true /* kubeletVisible */) {
```

Comment:
why are we passing an endpoints namespace twice to a function designed to get the named of referenced secrets?

Reply:
iiuc, this func ( `visitpvsecret`) entry is not required at all. I have commented out this part in the latest patch set. May be I am missing something here, please let me know if thats the case.

Reply:
correct, this change should be dropped... the field you are adding is not a reference to a secret, so there should be no changes here